src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -18,30 +32,107 @@
 
-    protected String profile = null;
-    protected GoGoDuckSubsetParameters subset = null;
-    protected UserLog userLog = null;
-    protected IndexReader indexReader = null;
+    @Autowired
+    ServletContext context;
+
+    private String profile = null;
+    private GoGoDuckSubsetParameters subset = null;
+    private UserLog userLog = null;
+    private IndexReader indexReader = null;
 
-    public GoGoDuckModule() {}
+    private Boolean postProcessingRequired = null;
+    private Boolean timeUnlimited = null;
+    private GoGoDuckConfig goGoDuckConfig;
 
-    public void init(String profile, IndexReader indexReader, String subset, UserLog userLog) {
+    public GoGoDuckModule() { }
+
+    public GoGoDuckModule(String profile, IndexReader indexReader, String subset, UserLog userLog, GoGoDuckConfig goGoDuckConfig) {
+        init(profile, indexReader, subset, userLog, goGoDuckConfig);
+    }
+
+    public void init(String profile, IndexReader indexReader, String subset, UserLog userLog, GoGoDuckConfig goGoDuckConfig) {
         this.profile = profile;
         this.indexReader = indexReader;
         this.subset = new GoGoDuckSubsetParameters(subset);
         this.userLog = userLog;
+        this.goGoDuckConfig = goGoDuckConfig;
     }
 
     public URIList getUriList() throws GoGoDuckException {
         return indexReader.getUriList(profile, TIME_FIELD, URL_FIELD, subset);
     }
 
+    public boolean postProcessingRequired() {
+        return postProcessingRequired.booleanValue();
+    }
+
+    public boolean isTimeUnlimited() {
+        return timeUnlimited.booleanValue();
+    }
+
+    public void setup(File file) {
+        boolean postProcessingRequired = false;
+        boolean timeUnlimited = true;
+        try {
+            String location = file.getAbsolutePath();
+            Set<NetcdfDataset.Enhance> enhanceMode = new HashSet<NetcdfDataset.Enhance>();
+            enhanceMode.add(NetcdfDataset.Enhance.ScaleMissing);
+            GridDataset gridDs = GridDataset.open(location, enhanceMode);
+            for (Variable var : gridDs.getNetcdfDataset().getVariables()) {
+                VariableDS vds = new VariableDS(var.getGroup(), var, true);
+                if (!postProcessingRequired && vds.hasScaleOffset()) {
+                    postProcessingRequired = true;
+                }
+
+                if (timeUnlimited && vds.isUnlimited()) {
+                    timeUnlimited = false;
+                }
+
+                if (postProcessingRequired && !timeUnlimited) {
+                    break;
+                }
+            }
+            this.postProcessingRequired = postProcessingRequired;
+            this.timeUnlimited = timeUnlimited;
+            gridDs.close();
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+            throw new GoGoDuckException(e.getMessage());
+        }
+    }
+
     public void postProcess(File file) {
```

Comment:
Perhaps move this to the gogoduck class to keep all actual file processing performed there and this class just for recording collection aggregation metadata